### ISSUE TITLE
Question: what happens with caching when using DSL alongside DAO

### ISSUE THREAD

**Issue by IgnatBeresnev** [JetBrains/Exposed#1593]
Hi! 

Recently I stumbled upon the same problem as described in https://github.com/JetBrains/Exposed/issues/914, where I have to upsert a value ([using this library](https://github.com/PerfectDreams/ExposedPowerUtils/blob/6004dc905372378d2297c3d19a51e091b9ad18e2/postgres-power-utils/src/main/kotlin/net/perfectdreams/exposedpowerutils/sql/upsert.kt#L22)) and then retrieve it by calling `Entity.findById` - it returns cached entity with outdated values. I've solved that by calling `Entity.removeFromCache` after doing an upsert, but that got me thinking

How safe is it to be using DSL queries alongside DAO? Is entity cache somehow updated/invalidated when DSL queries are performed? 

In other words, if I update an entity without explicitly committing/flushing, and then call some service which performs a DSL query to find this entity - will I see the updated value?

Should I stick to DSL only if I want to avoid possible cache-related problems?

Asking because DAO API is not as flexible, so I have to be doing DSL queries sometimes.
